app/helpers/protocols_io_helper.rb

```diff
@@ -0,0 +1,216 @@
+        td_strings = tr[0].scan(td_regex)
+        contents['data'][tr_counter] = []
+        td_strings.each do |td|
+          td_stripped = ActionController::Base.helpers.strip_tags(td[0])
+          contents['data'][tr_counter].push(td_stripped)
+        end
+      end
+      tables[table_counter.to_s]['contents'] = Base64.encode64(
+        contents.to_s.sub('=>', ':')
+      )
+      tables[table_counter.to_s]['name'] = nil
+    end
+    # return string_without_tables, tables
+    return tables, string_without_tables
+  end
+
+  def string_html_table_remove(description_string)
+    description_string.remove!("\n", "\t", "\r", "\f")
+    table_whole_regex = %r{(<table\b[^>]*>.*?<\/table>)}m
+    table_pattern_array = description_string.scan(table_whole_regex)
+    string_without_tables = description_string
+    table_pattern_array.each do |table_pattern|
+      string_without_tables = string_without_tables.gsub(
+        table_pattern[0],
+        t('protocols.protocols_io_import.comp_append.table_moved').html_safe
+      )
+    end
+    string_without_tables
+  end
+
+  def pio_eval_prot_desc(text, attribute_name)
+    case attribute_name
+    when 'before_start'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_BEFORESTART_RESERVED_LENGTH)
+    when 'warning'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_SAFETYWARNING_RESERVED_LENGTH)
+    when 'guidelines'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_GUIDELINES_RESERVED_LENGTH)
+    when 'publish_date'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_PBLDATE_RESERVED_LENGTH)
+    when 'vendor_name'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_VNDNAME_RESERVED_LENGTH)
+    when 'vendor_link'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_VNDLINK_RESERVED_LENGTH)
+    when 'keywords'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_KEYWORDS_RESERVED_LENGTH)
+    when 'link'
+      pio_eval_len(text, ProtocolsIoHelper::PIO_P_EXTLINK_RESERVED_LENGTH)
+    else
+      ''
+    end
+  end
+
+  def pio_eval_title_len(text)
+    text += ' ' if text.length < Constants::NAME_MIN_LENGTH
+    if text.length > Constants::NAME_MAX_LENGTH
+      text =
+        text[0..(Constants::NAME_MAX_LENGTH - PIO_TITLE_TOOLONG_LEN)] +
+        t('protocols.protocols_io_import.title_too_long')
+      @toolong = true
+    end
+    text
+  end
+
+  def pio_eval_len(text, reserved)
+    if text
+      text_end = reserved + @remaining - PIO_STEP_TOOLONG_LEN
+      if text.length - reserved > @remaining
+        text =
+          close_tags(text[0..text_end]) +
+          t('protocols.protocols_io_import.too_long')
+        @toolong = true
+      end
+      @remaining -= (text.length - reserved)
+      text
+    end
+  end
+
+  # Checks so that null values are returned as zero length strings
+  # Did this so views arent as ugly (i avoid using if present statements)
+  def not_null(attribute)
+    if attribute
+      attribute
+    else
+      ''
+    end
+  end
+
+  def close_tags(text)
```

Comment:
Maybe you could name this function differently, because it doesn't describe it's function properly.